File: core_migrations/backend/cpp/src/prompt.cpp
#include "core_migrations/backend/prompt.hpp"
#include "core_types/typing/types.hpp"
#include <memory>


namespace ct = core_types;
namespace cm = core_migrations;


void cm::prompt_notice(const std::string& notice)
{
    std::shared_ptr<ct::timestamptz> now = ct::timestamptz::utcnow();
    std::cout << "[NOTICE - " << now->to_string("%Y-%m-%dT%H:%M:%S") << "] " << notice << std::endl;
}


void cm::prompt_error(const std::string& error, const std::string& traceback)
{
    std::shared_ptr<ct::timestamptz> now = ct::timestamptz::utcnow();
    std::cout << "[ERROR - " << now->to_string("%Y-%m-%dT%H:%M:%S") << "] " << error << std::endl;
    std::cout << traceback << std::endl;
}


// void prompt_server_notice(const std::string& notice)
// {
//     now = datetime.now(timezone.utc)
//     sys.stdout.write(f"[SERVER - {now.isoformat()}] {diag.severity} - {diag.message_primary}\n")
// 
//     void prompt_notice(const std::string&);
//     void prompt_server_notice(const std::string&);
//     void prompt_error(const std::string&);
// }

File: core_migrations/scripts/_common/prompt.py
import sys
from typing import Any
import traceback
from datetime import\
    datetime,\
    timezone
import core_migrations.backend.cpp.backend_wrapper as bw


def prompt_yes_no(question) -> bool:
    """
    prompt [y/n] over question
    """
    valid = {"y": True, "n": False}

    while True:
        sys.stdout.write(f'[INPUT] {question} [y/n]: ')
        choice = input().lower()
        if choice in valid:
            return valid[choice]
        else:
            sys.stdout.write("[INPUT] Please respond with 'y' or 'n'.\n")


def prompt_notice(notice: str) -> None:
    bw.prompt_notice(notice)


def prompt_error(err: str) -> None:
    bw.prompt_error(err, str(traceback.format_exc()))


def log_notice(diag):
    now = datetime.now(timezone.utc)
    sys.stdout.write(f"[SERVER - {now.isoformat()}] {diag.severity} - {diag.message_primary}\n")
